optax/schedules/_schedule.py

```diff
@@ -465,7 +465,7 @@ def piecewise_interpolate_schedule(
 
   bounds = np.stack((0,) + boundaries)
   values = np.cumprod(np.stack((init_value,) + scales))
-  interval_sizes = bounds[1:] - bounds[:-1]
+  interval_sizes = np.maximum(bounds[1:] - bounds[:-1], 1)
```

Comment:
Maybe I'm reading this wrong but is this just checking that the keys of boundaries_and_scales is unique? Would it be better to check that explicitly and raise if not?

Reply:
Bounds are already passed as dictionary with bounds as the keys, so the already have to be unique. This is a sanity normalization for if the first bound is 0, resulting in 0 - 0 = 0